docs/user-guide/docker/deploying-in-docker.rst

```diff
@@ -0,0 +1,302 @@
+versions of Docker and Docker Compose. If the command doesn't work, you'll have to
+install Docker.
+See `Docker Installation <https://docs.docker.com/install/>`_ for details.
+
+
+Running the Rasa Server
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To run your AI assistant in production, configure your required
+:ref:`messaging-and-voice-channels` in ``credentials.yml``. If this file does not
+exist, create it using:
+
+.. code-block:: bash
+
+  touch credentials.yml
+
+Then edit it according to your connected channels.
+After, run the trained model with:
+
+.. code-block:: bash
+
+  docker run \
+    -v $(pwd)/models:/app/models \
+    rasa/rasa:latest-full \
+    run
+
+Command Description:
+
+  - ``-v $(pwd)/models:/app/models``: Mounts the directory with the trained Rasa model
+    in the container
+  - ``rasa/rasa:latest-full``: Use the Rasa image with the tag ``latest-full``
+  - ``run``: Executes the ``rasa run`` command. For more information see
+    :ref:`command-line-interface`.
+
+
+Using Docker Compose to Run Multiple Services
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To run Rasa together with other services, such as a server for custom actions, it is
+recommend to use `Docker Compose <https://docs.docker.com/compose/>`_.
+Docker Compose provides an easy way to run multiple containers together without
+having to run multiple commands.
+
+.. contents::
+   :local:
+   :depth: 2
+
+Start by creating a file called ``docker-compose.yml``:
+
+.. code-block:: bash
+
+  touch docker-compose.yml
+
+Add the following content to the file:
+
+.. code-block:: yaml
+
+  version: '3.0'
+  services:
+    rasa:
+      image: rasa/rasa:latest-full
+      ports:
+        - 5005:5005
+      volumes:
+        - ./:/app
+      command:
+        - run
+
+
+The file starts with the version of the Docker Compose specification that you
+want to use.
+Each container is declared as a ``service`` within the docker-compose file.
+The first service is the ``rasa`` service.
+
+The command is similar to the ``docker run`` command.
+The ``ports`` part defines a port mapping between the container and your host
+system. In this case it makes ``5005`` of the ``rasa`` service available on
+port ``5005`` of your host.
+This is the port of the :ref:`REST Channel <rest_channels>` interface of Rasa.
+
+.. note::
+
+    Since Docker Compose starts a set of Docker containers, it is no longer
+    possible to connect to the command line of a single container after executing the
+    ``run`` command.
+
+To run the services configured in your ``docker-compose.yml`` execute:
+
+.. code-block:: bash
+
+    docker-compose up
+
+
+
+
+.. _building-an-action-server-image:
+
+Deploying an Action Server Image
+********************************
+
+If you build an image that includes your action code and store it in a container registry, you can run it locally
+or as part of your Rasa X or Rasa Enterprise deployment, without having to move code between servers.
```

Comment:
any reason we just say Rasa X or Rasa Enterprise? It can be literally any deployment

Reply:
Good point